spec/controllers/concerns/billable_event_trackable_spec.rb

```diff
@@ -44,7 +53,33 @@
   end
 
   describe '#track_billing_events' do
-    it 'does not fail if SpReturnLog row already exists' do
+    let(:ial_context) { IalContext.new(ial: 2, service_provider: current_sp) }
+    it 'does not fail if SpReturnLog row already exists and ial 2' do
+      SpReturnLog.create(
+        request_id: request_id,
+        user_id: current_user.id,
+        billable: true,
+        ial: ial_context.ial,
+        issuer: current_sp.issuer,
+        requested_at: session_started_at,
+        returned_at: Time.zone.now,
+        profile_id: active_profile.id,
+        profile_verified_at: active_profile.verified_at,
+        profile_requested_issuer: active_profile.initiating_service_provider.issuer,
+      )
+
+      expect do
+        instance.track_billing_events
+      end.to_not(change { SpReturnLog.count }.from(1))
+      sp_return_log = SpReturnLog.last
+      expect(sp_return_log.profile_id).to eq active_profile.id
+      expect(sp_return_log.profile_verified_at).to eq active_profile.verified_at
+      expect(sp_return_log.profile_requested_issuer).
+        to eq active_profile.initiating_service_provider.issuer
```

Comment:
1. let's add a spec that shows the new attributes (`profile_id`, `profile_verified_at`, `profile_requested_issuer`) being added? this spec seems to only show that they do not change right?

2. let's save the assignment to make it clearer we're referencing the same row

```suggestion
      sp_return_log = SpReturnLog.create(
        request_id: request_id,
        user_id: current_user.id,
        billable: true,
        ial: ial_context.ial,
        issuer: current_sp.issuer,
        requested_at: session_started_at,
        returned_at: Time.zone.now,
        profile_id: active_profile.id,
        profile_verified_at: active_profile.verified_at,
        profile_requested_issuer: active_profile.initiating_service_provider.issuer,
      )

      expect do
        instance.track_billing_events
      end.to_not(change { SpReturnLog.count }.from(1))
      sp_return_log.reload
      expect(sp_return_log.profile_id).to eq active_profile.id
      expect(sp_return_log.profile_verified_at).to eq active_profile.verified_at
      expect(sp_return_log.profile_requested_issuer).
        to eq active_profile.initiating_service_provider.issuer
```